app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -546,6 +546,154 @@ class StateFragmentTest {
     }
   }
 
+  @Test
+  fun testStateFragment_loadExp_thirdState_hasSubmitButton() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      scrollToViewType(SUBMIT_ANSWER_BUTTON)
+      onView(withId(R.id.submit_answer_button)).check(
+        matches(withText(R.string.state_submit_button))
+      )
+      onView(withId(R.id.submit_answer_button)).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_changeConfiguration_thirdState_hasSubmitButton() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      rotateToLandscape()
+
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      scrollToViewType(SUBMIT_ANSWER_BUTTON)
+      onView(withId(R.id.submit_answer_button)).check(
+        matches(withText(R.string.state_submit_button))
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_thirdState_submitAnswer_submitButtonIsEnabled() {
```

Comment:
For this & other tests: should they be 'selectAnswer' instead of submit? Submission really only happens when the submit button is clicked.

Reply:
updated